backend_jobs/ingestion_removal/pipeline_lib/firestore_database.py

```diff
@@ -0,0 +1,210 @@
+
+        Args:
+            element: the lower limit for querying the database by the random field.
+            image_provider: The image provider from whom we are remove the images.
+            pipeline_run: The image pipeline_run from whom we are remove the images.
+
+        Yields:
+            A tuple containing of a parents image id and the image provider/pipeline run,
+            e.g. if in this functions input was an image provider- in the tuple we
+            return the pipeline run of the deleted document.
+        """
+        _valedate_one_arg(image_provider, pipeline_run)
+        # The lower limit for querying the database by the random field.
+        random_min = element * constance.RANGE_OF_BATCH
+        # The higher limit for querying the database by the random field.
+        random_max = random_min + constance.RANGE_OF_BATCH
+        if image_provider:
+            query = self.db\
+                .collection_group(database_schema.COLLECTION_IMAGES_SUBCOLLECTION_PIPELINE_RUNS)\
+                .where(database_schema.COLLECTION_IMAGES_SUBCOLLECTION_PIPELINE_RUNS_FIELD_PROVIDER_ID, u'==', image_provider)\
+                .where(database_schema.COLLECTION_IMAGES_SUBCOLLECTION_PIPELINE_RUNS_FIELD_RANDOM, u'>=', random_min)\
+                .where(database_schema.COLLECTION_IMAGES_SUBCOLLECTION_PIPELINE_RUNS_FIELD_RANDOM, u'<', random_max)\
+                .stream()
+        else:  # if pipeline_run.
+            query = self.db\
+                .collection_group(database_schema.COLLECTION_IMAGES_SUBCOLLECTION_PIPELINE_RUNS)\
+                .where(database_schema.COLLECTION_IMAGES_SUBCOLLECTION_PIPELINE_RUNS_FIELD_PIPELINE_RUN_ID, u'==', pipeline_run)\
+                .where(database_schema.COLLECTION_IMAGES_SUBCOLLECTION_PIPELINE_RUNS_FIELD_RANDOM, u'>=', random_min)\
+                .where(database_schema.COLLECTION_IMAGES_SUBCOLLECTION_PIPELINE_RUNS_FIELD_RANDOM, u'<', random_max)\
+                .stream()
+        for doc in query:
+            doc_dict = doc.to_dict()
+            parent_image_id = doc_dict[
+                database_schema.COLLECTION_IMAGES_SUBCOLLECTION_PIPELINE_RUNS_FIELD_PARENT_IMAGE_ID]
+            # If we remove by provider we will return the pipeline run to remove if this pipeline run
+            # is provider specific.
+            if image_provider:
+                pipeline_run = doc_dict[
+                    database_schema.COLLECTION_IMAGES_SUBCOLLECTION_PIPELINE_RUNS_FIELD_PIPELINE_RUN_ID]
+                yield (parent_image_id, pipeline_run)
+            # If we remove by pipeline_run we will return the image provider to remove if the provider
+            # is pipeline run specific.
+            else:
+                image_provider = doc_dict[
+                    database_schema.COLLECTION_IMAGES_SUBCOLLECTION_PIPELINE_RUNS_FIELD_PROVIDER_ID]
+                yield (parent_image_id, image_provider) 
```

Comment:
all these queries are not needed as far as I know. we just need the parent_image_id